examples/todo-typescript/src/client/MainPage.tsx

```diff
@@ -0,0 +1,88 @@
+import './Main.css';
+import React, { useEffect, FormEventHandler } from 'react';
+import logout from '@wasp/auth/logout.js';
+import useAuth from '@wasp/auth/useAuth.js';
+import { useQuery } from '@wasp/queries'; // Wasp uses a thin wrapper around react-query
+import getTasks from '@wasp/queries/getTasks';
+import createTask from '@wasp/actions/createTask';
+import updateTask from '@wasp/actions/updateTask';
+import waspLogo from './waspLogo.png';
+import { Task } from './types'
+
+export function MainPage() {
+  const { data: user } = useAuth();
+  const { data: tasks, isLoading, error } = useQuery<unknown, Task[]>(getTasks);
+
+  useEffect(() => {
+    console.log(user);
+  }, [user]);
+
+  if (isLoading) return 'Loading...';
+  if (error) return 'Error: ' + error;
+
+  return (
+    <main>
+      <img src={waspLogo} alt='wasp logo' />
+      <h1>
+        {user.username}
+        {`'s tasks :)`}
+      </h1>
+      <NewTaskForm />
+      {tasks && <TasksList tasks={tasks} /> }
+      <button onClick={logout}> Logout </button>
+    </main>
+  );
+};
+
+function Todo({ id, isDone, description }: Task) {
+  const handleIsDoneChange = async (event: FormEventHandler<HTMLInputElement>) => {
+    try {
+      await updateTask({
+        taskId: id,
+        isDone: event.currentTarget.checked,
+      });
+    } catch (err: any) {
+      window.alert('Error while updating task ' + err?.message);
+    }
+  };
+
+  return (
+    <li>
+      <input type='checkbox' id={id.toString()} checked={isDone} onChange={handleIsDoneChange} />
+      <span>{description}</span>{' '}
+    </li>
+  );
+};
+
+function TasksList({tasks}: { tasks: Task[] }) {
+  if (tasks.length === 0) return <p>No tasks yet.</p>;
+  return (
+    <ol className='tasklist'>
+      {tasks.map((tsk, idx) => (
+        <Todo {...tsk} key={idx} />
+      ))}
+    </ol>
+  );
+};
+
+function NewTaskForm() {
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
Same thing applies: https://github.com/wasp-lang/wasp/pull/886#discussion_r1069416981.

You can ditch the `React` import.

Reply:
🚀